systems/src/overhead/mod.rs

```diff
@@ -0,0 +1,235 @@
+#[derive(PartialEq)]
+pub enum OnOffPushButtonState {
+    On,
+    Off,
+}
+
+pub struct OnOffPushButton {
+    state: OnOffPushButtonState,
+    fault: bool,
+    available: bool,
+}
+impl OnOffPushButton {
+    pub fn new_on() -> OnOffPushButton {
+        OnOffPushButton {
+            state: OnOffPushButtonState::On,
+            fault: false,
+            available: false,
+        }
+    }
+
+    pub fn new_off() -> OnOffPushButton {
+        OnOffPushButton {
+            state: OnOffPushButtonState::Off,
+            fault: false,
+            available: false,
+        }
+    }
+
+    pub fn set(&mut self, value: bool) {
+        self.state = if value {
+            OnOffPushButtonState::On
+        } else {
+            OnOffPushButtonState::Off
+        };
+    }
+
+    pub fn set_fault(&mut self, fault: bool) {
+        self.fault = fault;
+    }
+
+    pub fn turn_on(&mut self) {
+        self.state = OnOffPushButtonState::On;
+    }
+
+    pub fn turn_off(&mut self) {
+        self.state = OnOffPushButtonState::Off;
+    }
+
+    pub fn set_available(&mut self, available: bool) {
+        self.available = available;
+    }
+
+    pub fn shows_available(&self) -> bool {
+        self.available
+    }
+
+    pub fn has_fault(&self) -> bool {
+        self.fault
+    }
+
+    pub fn is_on(&self) -> bool {
+        self.state == OnOffPushButtonState::On
+    }
+
+    pub fn is_off(&self) -> bool {
+        self.state == OnOffPushButtonState::Off
+    }
+}
+
+#[derive(PartialEq)]
+pub enum NormalAltnPushButtonState {
+    Normal,
+    Altn,
+}
+
+pub struct NormalAltnPushButton {
+    state: NormalAltnPushButtonState,
+    fault: bool,
+}
+impl NormalAltnPushButton {
```

Comment:
What's a NormalAltnPushButton? The name seems unclear to me, but it seems to refer to the fault light or something? Does this mirror an Asobo XML name?

Reply:
It's used for the AC ESS FEED button. Normal for when the AC ESS FEED should automatically switch between AC BUS 1 and 2; ALTN for when AC BUS 2 is manually selected.

![image](https://user-images.githubusercontent.com/1074786/106155683-3bfc5480-6181-11eb-9644-23eaa2081c93.png)